### PR TITLE
Refactor PianoRoll code

### PR DESCRIPTION
**Pull request by Wallacoloo**
With one exception\*, this PR changes none of the actual behavior of the Piano Roll, but just makes the code easier to understand.

\* The exception is that previously when you used the mouse wheel to change the volume/panning of some notes, the "Volume: nn%" hover-text would be displayed only if you were editing a single note. Now it will display as long as all the notes being edited have the same volume (or panning, if that's the attribute you're editing). This is often useful when changing the volume/panning of a chord.

**Besides that, the internal changes are**:
* Restructure parts to avoid redundant or duplicated code (especially where string constants are involved).
* Formatting/tab-alignment fixes.
* Better const-correctness.
* Remove 'inline' attribute where not applicable (inline is only needed when defining functions in a header, as it prevents the linker from issuing multiple-definition errors. It is otherwise useless).
* Use C++11 range-based for loops (`for (Note * note : notes) {}`) as opposed to explicit iterators or Qt's `foreach(Note * note, notes) {}` when possible.
* Rename `copy_to_clipboard()` -> `copyToClipboard()`
* Change `void getSelectedNotes(NoteVector & selected_notes)` to `NoteVector getSelectedNotes()` - this function previously used a write-back parameter, appending notes to `selected_notes`. Now it just creates a new vector, populates it, and returns that. Performance differences should be negligible, and return values should be prefered over write-back parameters when possible.

======
There's still a lot of work to be done in cleaning up that file, but this is a start.

### CODE REVIEW

File: src/gui/editors/PianoRoll.cpp [hunk 1]
```diff
@@ -728,8 +766,8 @@ inline void PianoRoll::drawNoteRect(QPainter & p, int x, int y,
 
 
 
-inline void PianoRoll::drawDetuningInfo( QPainter & _p, Note * _n, int _x,
-								int _y )
+inline void PianoRoll::drawDetuningInfo( QPainter & _p, const Note * _n, int _x,
```
Comment (curlymorphic) [inline src/gui/editors/PianoRoll.cpp]: There appears to be an inline, that has slipped through the net.

**Comment by curlymorphic** [comment]
Nice work @Wallacoloo. 

IIRC this file was in desperate need of a clean up.

I have read various comments about the use of inline statements, that I have never verified, but I am of the school of thought that they are overused in our code base, I can't see there was ever a need for them in UI code. 

My only other comment, that is more  a note for future tidying of our code base and no reflection on your work, is the inconsistent use of whitespace with brackets. I am aware the compulsory use has been dropped from our requirements, but looking over this file it is apparent that there is a lot of mixed use within the same file.

In conclusion, there is just the single inline statement, and I feel this is good to merge

**Comment by Wallacoloo** [comment]
Good catch @curlymorphic! I revised commit 2e11112 to include the removal of that inline attribute. Waiting for Travis to complete & then this can be merged.
